File: servers/Services/AcasService/Models/RegradingRequest.cs
using System.ComponentModel.DataAnnotations;

namespace AcasService.Models;

public class RegradingRequest
{
    [Key]
    public string Id { get; set; } = string.Empty;
    
    private string ExaminationId { get; set; } = string.Empty;
    
    private string ProblemId { get; set; } = string.Empty;
    
    private string SubmissionId { get; set; } = string.Empty;
    
    [Required]
    public string StudentId { get; set; } = string.Empty;
    
    [Required]
    public string Reason { get; set; } = string.Empty;
    
    public DateTime CreatedDate { get; set; }
    
    [Required]
    public RegradingRequestStatus Status { get; set; }
    
    public string LecturerNote { get; set; } = string.Empty;
    
    public DateTime HandledDate { get; set; }
}

public enum RegradingRequestStatus
{
    PENDING,
    APPROVED,
    REJECTED,
    CANCELED
}

File: web_app/types/regrading-request.ts
export type RegradingRequestStatus = "PENDING" | "APPROVED" | "REJECTED" | "CANCELED";

export interface SubmissionLite {
  id: string;
  studentId: string;
  examId: string;
  problemId: string;
  finalScore: number;
  submittedDate: string;
}

export interface RegradingRequest {
  id: string;
  examinationId: string;
  submissionId: string;
  studentId: string;
  studentName: string;
  studentEmail: string;
  reason: string;
  imageUrls: string[];
  createdDate: string;
  status: RegradingRequestStatus;
  statusName: string;
  lecturerNote: string;
  handledDate: string | null;
  submission: SubmissionLite | null;
}

export interface PagedRegradingRequests {
  items: RegradingRequest[];
  totalCount: number;
  pageIndex: number;
  pageSize: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}

export interface CreateRegradingRequestPayload {
  examinationId: string;
  submissionId: string;
  imageUrls: string[];
  reason: string;
}

export interface HandleRegradingRequestPayload {
  lecturerNote: string;
}
